hs_core/models.py

```diff
@@ -2271,6 +2309,40 @@ def supports_delete_folder_on_zip(self, original_folder):
         """Check if resource allows the original folder to be deleted upon zip."""
         return True
 
+    @property
+    def storage_type(self):
+        if not self.is_federated:
+            return 'local'
+        userpath = '/' + os.path.join(
+            getattr(settings, 'HS_USER_IRODS_ZONE', 'hydroshareuserZone'),
+            'home',
+            getattr(settings, 'HS_LOCAL_PROXY_USER_IN_FED_ZONE', 'localHydroProxy'))
+        if self.resource_federation_path == userpath:
+            return 'user'
+        else:
+            return 'federated'
+
+    def is_folder(self, folder_path):
+        """Determine whether a given path (relative to resource root, including /data/contents/)
+           is a folder or not. Returns False if the path does not exist.
+        """
+        path_split = folder_path.split('/')
+        while path_split[-1] == '':
+            path_split.pop()
+        dir_path = u'/'.join(path_split[0:-1])
+
+        # handles federation
+        irods_path = os.path.join(self.root_path, dir_path)
```

Comment:
How does this handle federation?  Does root_path take into account federation?  Can we do the same above in hs_composite_resource/models.get_folder_aggregation_type_to_set()?